src/span_context.js

```diff
@@ -11,55 +11,21 @@
  * <trace_id, span_id, sampled> tuple).
  */
 export default class SpanContext {
-
     /**
-     * Sets a key:value pair on this SpanContext that also propagates to future
-     * children of the associated Span.
-     *
-     * setBaggageItem() enables powerful functionality given a full-stack
-     * opentracing integration (e.g., arbitrary application data from a web
-     * client can make it, transparently, all the way into the depths of a
-     * storage system), and with it some powerful costs: use this feature with
-     * care.
-     *
-     * IMPORTANT NOTE #1: setBaggageItem() will only propagate baggage items to
-     * *future* causal descendants of the associated Span.
-     *
-     * IMPORTANT NOTE #2: Use this thoughtfully and with care. Every key and
-     * value is copied into every local *and remote* child of the associated
-     * Span, and that can add up to a lot of network and cpu overhead.
+     * Creates a new copy of this SpanContext, and a key:value pair as baggage on this span.
      *
      * @param {string} key
      * @param {string} value
-     */
-    setBaggageItem(key, value) {
-        if (API_CONFORMANCE_CHECKS) {
-            if (arguments.length !== 2) {
-                throw new Error('Invalid number of arguments');
-            }
-        }
-        if (this._imp) {
-            this._imp.setBaggageItem(key, value);
-        }
-    }
-
-    /**
-     * Returns the value for a baggage item given its key.
      *
-     * @param  {string} key
-     *         The key for the given trace attribute.
-     * @return {string}
-     *         String value for the given key, or undefined if the key does not
-     *         correspond to a set trace attribute.
+     * @return {Object} - returns the new SpanContext with key:value pair set
+     * as baggage on the new context.
      */
-    getBaggageItem(key) {
+    withBaggageItem(key, value)  {
```

Comment:
withBaggageItem is actually an implementation detail. Does it need to be in the API?  From [this](https://github.com/opentracing/opentracing-javascript/blob/master/src/span.js#L34) it looks like the API method `context()` is always retrieving the current context from the `impl` and wrapping it, so `withBaggageItem` can be handled completely internally to the span implementation.